src/main/java/org/jabref/gui/entryeditor/LatexCitationsTabViewModel.java

```diff
@@ -0,0 +1,138 @@
+package org.jabref.gui.entryeditor;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.ReadOnlyListWrapper;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+import org.jabref.gui.AbstractViewModel;
+import org.jabref.gui.util.BackgroundTask;
+import org.jabref.gui.util.TaskExecutor;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.texparser.DefaultTexParser;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.texparser.Citation;
+import org.jabref.model.texparser.TexParserResult;
+import org.jabref.preferences.PreferencesService;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LatexCitationsTabViewModel extends AbstractViewModel {
+
+    enum Status {
+        IN_PROGRESS,
+        CITATIONS_FOUND,
+        NO_RESULTS,
+        ERROR,
+        INACTIVE
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LatexCitationsTabViewModel.class);
+    private static final String TEX_EXT = ".tex";
+    private final BibDatabaseContext databaseContext;
+    private final PreferencesService preferencesService;
+    private final TaskExecutor taskExecutor;
+    private final StringProperty entryKey;
+    private final ObservableList<Citation> citationList;
+    private final ObjectProperty<Status> status;
+    private final StringProperty searchError;
+    private Future<?> searchTask;
+
+    public LatexCitationsTabViewModel(BibDatabaseContext databaseContext, PreferencesService preferencesService,
+                                      TaskExecutor taskExecutor) {
+        this.databaseContext = databaseContext;
+        this.preferencesService = preferencesService;
+        this.taskExecutor = taskExecutor;
+        this.entryKey = new SimpleStringProperty(null);
+        this.citationList = FXCollections.observableArrayList();
+        this.status = new SimpleObjectProperty<>(Status.INACTIVE);
+        this.searchError = new SimpleStringProperty(null);
+    }
+
+    public void init(BibEntry entry) {
+        cancelSearch();
+
+        entryKey.set(entry.getCiteKeyOptional().orElse(null));
```

Comment:
Please don't use `orElse(null)`. The point of optionals is exactly to remove null values (and thus prevent NPEs). Here, I would propose the following code:
````java
Optional<String> citeKey = entry.getCiteKeyOptional();
if (citeKey.isPresent()) {
   startSearch(citeKey.get());
} else {
   searchError.set(Localization.lang("Selected entry does not have an associated BibTeX key."));
   status.set(Status.ERROR);
}
```